### DISCUSSION TITLE
Next Js Image component

### DISCUSSION THREAD

**Discussion by ghost** [rpearce/react-medium-image-zoom#307] [Q&A] [answered]
Thanks for making this amazing package. But how can I use it with Image component in next js. Please show me an example.

**Comment by rpearce** [comment, marked as answer]
@studywithdesign I rarely use NextJS, so I'm a bit unhelpful there, but [this issue and comment might be helpful](https://github.com/rpearce/image-zoom/issues/281#issuecomment-1111246126).

**Comment by rpearce** [comment]
Update: You should be able to wrap your nextjs image component in the `<Zoom></Zoom>` component, and it should work fine.

**Comment by Tobias-Ertel** [comment]
How can I use it for images in a Markdown File (in a Next.js blog)? I already tried to warp the hole container in a <Zoom></Zoom> component but it always just works for the first images. At the other images only the magnifying glass icon appears but its not possible to zoom the Image. Does any one else has the same problem?

**Reply by rpearce** [reply]
@Letre2 I've never used a nextjs blog, so please forgive my ignorance around that.

It sounds like you're asking a similar question as the person in https://github.com/rpearce/react-medium-image-zoom/issues/331.

My answer there was:

> This project has never purposefully supported zooming more than 1 image inside of a <Zoom> component.
>
> Instead, each instance of an image should be wrapped in a <Zoom> component.

Does this help answer your question? Do you have an example project to share that I could look at to understand better?

**Reply by rpearce** [reply]
I understand this question better now. I don't currently have an answer, as each image needs to be wrapped in this component, and you're working in Markdown... 🤔

**Reply by Tobias-Ertel** [reply]
Yes, my solution was to switch to MDX and then it’s possible to wrap everyone in a Zoom Tag, since it’s possible to import packages with MDX. But thanks so much for your response!

**Reply by korompaiistvan** [reply]
for people stumbling on this. the solution is exactly as @Letre2 says. switch to a markdown renderer/serializer that allows you to define custom components for images. more specific advice depends on a lot of things, but looking around MDX related solutions will usually help, as custom component support is the main distinction between mdx and markdown
